go-controller/pkg/ovn/egressip.go

```diff
@@ -2099,11 +2101,39 @@ func (oc *Controller) checkEgressNodesReachability() {
 }
 
 func (oc *Controller) isReachable(node *egressNode) bool {
-	for _, ip := range node.mgmtIPs {
-		if dialer.dial(ip) {
-			return true
+	var maxRetryCnt int
+	var retryTimeOut, initialRetryTimeOut time.Duration
+
+	numMgmtIPs := len(node.mgmtIPs)
+	if numMgmtIPs == 0 {
+		return false
+	}
+
+	switch oc.eIPC.egressIPTotalTimeout {
+	// Check if we need to do node reachability check
+	case 0:
+		return true
+	case 1:
+		maxRetryCnt = 2
+		initialRetryTimeOut = time.Duration(700/numMgmtIPs) * time.Millisecond
+		retryTimeOut = time.Duration(100/numMgmtIPs) * time.Millisecond
+	default:
+		maxRetryCnt = oc.eIPC.egressIPTotalTimeout
+		initialRetryTimeOut = time.Duration(900/numMgmtIPs) * time.Millisecond
+		retryTimeOut = initialRetryTimeOut
+	}
+
+	timeout := initialRetryTimeOut
+	for retryCnt := 0; retryCnt < maxRetryCnt; retryCnt++ {
```

Comment:
Don't count the number of retries and compare against maxretry. Make the loop time based and end after  oc.eIPC.egressIPTotalTimeout:
~~~
	endTime := time.Now().Add(time.Second * time.Duration(oc.eIPC.egressIPTotalTimeout))
	for time.Now().Before(endTime) {
~~~